macros/src/lib.rs

```diff
@@ -992,11 +992,11 @@ fn mksym(string: &str, tag: &str, is_log_statement: bool) -> TokenStream2 {
         format_ident!("S")
     };
     quote!(match () {
-        #[cfg(target_arch = "x86_64")]
+        #[cfg(feature = "unstable-test")]
```

Comment:
~~this `cfg` however will end in the code emitted by the logging macros, `write!`, `intern!`, etc. I'm not sure if that's the behavior we want.~~

~~The other option is make the defmt macros *always* emit the `fetch_add_string_index` version if its "unstable-test" feature is enabled.~~

EDIT: OK, after reading the rest of the PR. I think this should be:
``` rust
if cfg!(feature = "unstable-test") {
    quote!(/* .. */) // fetch add string version
} else {
    quote!(/* .. */) // linker version
}
```